### ISSUE TITLE
How to test on pc?

### ISSUE THREAD

**Issue by SaturnusDJ** [jursonovicst/gradient#3]
I ran the video file on pc with a R9 280 graphics card sending to an LG 24UD58-B connected via DisplayPort @ UHD / 3840x2160 @ 60Hz. There is a visible difference between the 2 bands (especially in the first third of the grey gradient) but when changing the color depth in Radeon Settings to 8 bpc or 6 bpc, this difference is still visible.

**Comment by jursonovicst** [comment]
Hi SaturnusDJ,

According to the [specification](https://www.lg.com/uk/monitors/lg-24UD58-B) of this monitor, it does not have a true 10 bit panel, instead, it implements [adaptive frame rate control](https://en.wikipedia.org/wiki/Frame_rate_control) to trick your eyes.

Because this implementation is vendor dependent, I am not sure, what they do and how they downsample the 10 bit source and how they build intermediate colors. Sorry, but I have not tested such a model yet. What I can recommend is:

- run the video in 1:1 aspect ratio, make sure that the player does not scale it,
- turn off all "Image Enhancement" features, like SUPEr rESOLUtIOn+, DFC, FreeSync, Black Stabilizer,
- test these [dev sequences,](
https://github.com/jursonovicst/gradient/tree/develop/mobile_test_sequence) they have wider stripes, which can be observed easier.

Tell me about your results.

**Comment by SaturnusDJ** [comment]
Thanks.

Sakkura @ http://www.tomshardware.co.uk/forum/id-3320965/true-10bit-8bit-frc.html mentions it happens by dithering in the frames.

When using the dev files:
The images shows 25 stripes in each band.
The videos show 25 stripes in the first band and indeed about 100 in the second band.

The videos were tried in Windows Media Player with Shark007 codecs installed and in VLC Player.

I am not convinced yet. Why would the image not display in 10-bit. Are the video codecs maybe using a dithering technique instead of the monitor?

**Comment by SaturnusDJ** [comment]
Now that I asked the above question I remember something which might point to 'yes'... When playing the video at an LG 24EB23 there also is a clear difference between both bands.

**Comment by jursonovicst** [comment]
Hi @SaturnusDJ,

The images are 16bit PNG images, and I had the same problem on Samsung TV sets, where the photo viewer application did not support more than 8 bit for presenting photos, however the video player app succeeded with 10 bit videos.

So at the end, you were able to confirm the 10 bit video support on the LG 24UD58-B and on LG 24EB23 as well, weren't you? May I close this case?

**Comment by SaturnusDJ** [comment]
Except...the LG 24EB23 is a 8-bit panel.

**Comment by SaturnusDJ** [comment]
Can you please generate FullHD (1920x1080, or even 1920x1200) versions of the test? Which those formats we might be able to rule out video codec enhancements and/or pixel compression on the LG 24EB23 which in turn, hopefully, result in the 10 and 8-bit strokes to look the same on that display.

**Comment by jursonovicst** [comment]
[Here](https://github.com/jursonovicst/gradient/tree/develop/test_sequences) they are... tell me about your results.

**Comment by SaturnusDJ** [comment]
Thanks!
Although they look a lot different from the UHD versions, I ran the 60-80 and 80-100 x265 versions on the LG 24EB23 (1920x1200 8-bit) and the result is: 10-bit looks smoother.

In other words:
a) Either the video decoders (LAV in this case) is enhancing the video.
or
b) The video's are not really 10-bit.

I have verified with GIMP that the PNG-images are 10-bit by displaying the colour values as 0-65k range with shows a much higher resolution in the 10-bit band.

The video analysed with Shark007 settings application (function likely from https://mediaarea.net/en/MediaInfo) shows the video is 10-bit YUV 4:2:0.

Assuming that is correct...I am no color bit depth expert, so correct me if I am wrong but wouldn't 4:4:4 be better in the case of quality testing files?
https://www.rtings.com/tv/learn/chroma-subsampling
https://www.youtube.com/watch?v=yinOtVTWJFY

**Comment by SaturnusDJ** [comment]
Or switch from YUV to RGB if the codecs allow.

Some nice visual examples in this (Dutch) article: https://www.homecinemamagazine.nl/2017/02/420-chroma-subsampling-uitgelegd/

**Comment by jursonovicst** [comment]
You are right, I developed the first sequences to test UHD TV sets, but they do support only yuv4:2:0. Since you are testing on a PC, let me see, what can I do with color subsampling and rap pixel format. I will create new sequences for you tonight, thanks for helping with testing...

**Comment by SaturnusDJ** [comment]
Great! No problem, thanks for dedication to the repo.

Somehow I have the feeling 10-bit isn't going to happen to my computer, but we will see. We already have proven that it is good to dig deeper than just to assume 10-bit is working because Windows says so. 👌

**Comment by jursonovicst** [comment]
Try these: https://github.com/jursonovicst/gradient/tree/develop/test_sequences/1920x1080

Now x265 with
 - lossless encoding
 - full color range

I have realized, that without these settings, the initial RGB->YUV conversion did some [color range cropping](https://stackoverflow.com/questions/47900346/ffmpeg-color-range-cropped-to-15-235-in-rgb). Using VLC, I confirm the right full color range (0...255) on my 8 bit display, but I cannot test it on 10 bit display till Tuesday. Please try.

There are also videos with yuv420, yuv444, and rgb pixel format.

Make sure, that your player [will not crop](https://webtrickz.com/how-to-fix-washed-out-colors-issue-in-vlc-player/) the color range again at YUV->RGB conversion

**Comment by jursonovicst** [comment]
Do you see now any difference between yuv420, yuv444, and rgb?

**Comment by SaturnusDJ** [comment]
YUV444 and RGB are definitely better quality than YUV420. I used Windows Magnifier though to see this to spare myself minute long testing.

YUV420:
![mag420](https://user-images.githubusercontent.com/14320018/47287478-a7ded280-d5f2-11e8-8599-b77b9f93172a.png) (Notice the compression above the 'l' in the second 'should.')

YUV444:
![mag444](https://user-images.githubusercontent.com/14320018/47287489-af05e080-d5f2-11e8-9dd9-5859df59b653.png)

RGB:
![magrgb](https://user-images.githubusercontent.com/14320018/47287496-b3ca9480-d5f2-11e8-9d03-b3d922590163.png)

Again something strange: I used the 60-80 gradients, yet the YUV444 is clearly darker overall compared to the other two. Therefore I cannot check (technically) if RGB is better than YUV444.

Proof:
![mag444rgbdiff](https://user-images.githubusercontent.com/14320018/47287881-06588080-d5f4-11e8-81f6-bb14b8eff1e2.png)
On top YUV444, RGB below.

**Comment by SaturnusDJ** [comment]
If you still like me to test something, the UHD monitor stays until Tuesday. It's a bit too small.
Maybe getting a new one soon, or maybe in 2 years.
